sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/XmlExternalEntityShouldNotBeParsed.cs

```diff
@@ -225,5 +224,17 @@ private static bool IsAllowedValueForXmlTextReader(object constantValue)
                 symbol is IMethodSymbol methodSymbol &&
                 methodSymbol.ReturnType.IsAny(UnsafeXmlResolvers);
         }
+
+        private class TrackersHolder
```

Comment:
I think we can use a readonly struct in this case to avoid a new object allocation.